content/blog/2020-08-19-train-a-sequence-model-with-poutyne/index.en.md

```diff
@@ -0,0 +1,613 @@
+bidirectional approach (bidirectional LSTM). What it means is that instead of _simply_ viewing the sequence from the start to the end, we
+also train the model to see the sequence from the end to the start. It's important to state that the two directions are
+not shared, meaning that we _see_ the sequence in one direction at the time, but we gather the information from both directions into the 
+fully connected layer. That way, our model can get insight from both directions.
+
+Instead of using only one layer, let's use a bidirectional bi-LSTM, which means that we use two layers of hidden state for each direction.
+
+So, let's create the new LSTM and fully connected network.
+
+```python
+dimension = 300
+num_layer = 2
+bidirectional = True
+
+lstm_network = nn.LSTM(input_size=dimension,
+                       hidden_size=dimension,
+                       num_layers=num_layer,
+                       bidirectional=bidirectional,
+                       batch_first=True)
+
+input_dim = dimension * 2 #since bidirectional
+
+fully_connected_network = nn.Linear(input_dim, tag_dimension)
+
+full_network_bi_lstm = FullNetWork(lstm_network, fully_connected_network)
+```
+ 
+### Training
+ 
+```python
+exp_bi_lstm = Experiment("./", full_network_bi_lstm, device=cuda_device, optimizer=optimizer,
+                 loss_function=cross_entropy, batch_metrics=["acc"])
+exp_bi_lstm.train(train_loader, valid_generator=valid_loader, epochs=epoch_number)
+```
+
+### Results
+Here are our validation results for the last epoch of the larger model. On the validation dataset,
+we can see that we obtain a marginal gain of around `0.3` for the accuracy over the previous one. Not much of an improvement.
+
+|   Model  | Bidirectional bi-LSTM |
+|:--------:|:------------------:|
+|   Loss   |    0.0050          |
+| Accuracy |    99.8594         |
+
+But now that we have our two trained models, let's use the test set as a final and **unique** step for evaluating their performance.
+
+```python
+exp.test(test_loader)
+exp_bi_lstm.test(test_loader)
+```
+
+The next table presents the results of the bidirectional bi-LSTM with two layers and the previous model (LSTM with one layer).
+
+|   Model  | LSTM one layer | Bidirectional bi-LSTM |
+|:--------:|:--------------:|:------------------:|
+|   Loss   |     0.0152     |    **0.0050**      |
+| Accuracy |     99.5758    |    **99.8550**     |
+
+We see similar validation results for both models. Also, we still see a little improvement in accuracy and total loss for the larger model. Considering that we only improved by around 0.3, one can
+argue that it's only a matter of seed. To test the robustness of our approach, we could train our model multiple times
+using different seeds and report the mean and standard deviation of each metric over all experiments rather than the result of a single training. Let's try something else. 
+
+### Zero Shot Evaluation
```

Comment:
The point of languages has to be addressed. You say you are training with french pre-trained embeddings, yet later refer to your training set as in English.

How do the embeddings predict other languages than French, say Russian and mexican ? The model is trained with respect to the french embeddings' intrinsic geometry, you should address that most of the errors on these languages can be attributed to the embeddings losing much of their meaning because of input language disparity.